gson/src/main/java/com/google/gson/stream/JsonWriter.java

```diff
@@ -490,6 +490,25 @@ public JsonWriter value(Boolean value) throws IOException {
     return this;
   }
 
+  /**
+   * Encodes {@code value}.
+   *
+   * @param value a finite value. May not be {@link Float#isNaN() NaNs} or {@link Float#isInfinite()
```

Comment:
Might make sense to remove this "May not be NaN or Infinity" (also for `value(double)` and `value(Number)`) because it is misleading since this only applies if the writer is non-lenient and is already covered by the `@throws IllegalArgumentException`.

Feel free to only consider that a suggestion; I am not a member of this project.

Reply:
See #2132 for some wordsmithing here in addition to adding the`@Overrides` in `JsonTreeWriter`.

Reply:
Thanks! Could you please adjust the documentation for `value(Number)` in a similar way?

Reply:
Done.

Reply:
If there are other issues, please comment inline on #2132 .